[CI] Solaris 11x64 rsyslog failed — rsyslog solaris11x64 rsyslog — BUILD FAILED: failed 'gmake -j2 ...' (failure)

## Solaris 11x64 rsyslog CI failure

**Summary:** BUILD FAILED: failed 'gmake -j2 ...' (failure)
**Builder:** rsyslog solaris11x64 rsyslog
**Build:** https://build.rsyslog.com/#/builders/77/builds/7132
**Revision:** 993068f961ee1f22b286e754619b300e20bb8f1f
**Worker:** slave-solaris11x64

### Steps

- 0: worker_preparation — success

- 1: set http proxy in git config — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/1/logs/stdio) 

- 2: git — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/2/logs/stdio) 

- 3: shell — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/3/logs/stdio) 

- 4: cleanup — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/4/logs/stdio) 

- 5: cleanup dependencies — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/5/logs/stdio) 

- 6: git branch information — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/6/logs/stdio) 

- 7: building librelp dependency — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/7/logs/stdio) 

- 8: building libfastjson dependency — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/8/logs/stdio) 

- 9: autoreconf for SunStudio Compiler — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/9/logs/stdio) 

- 10: shell_1 — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/10/logs/stdio) [config.log](https://build.rsyslog.com/#/builders/77/builds/7132/steps/10/logs/config_log) 

- 11: build with SunStudio — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/11/logs/stdio) 

- 12: shell_2 — failure — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/12/logs/stdio) [test-suite.log](https://build.rsyslog.com/#/builders/77/builds/7132/steps/12/logs/test-suite_log) 

- 13: gathering make check logs — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/13/logs/stdio) 

- 14: shell_3 — success — [stdio](https://build.rsyslog.com/#/builders/77/builds/7132/steps/14/logs/stdio)

--- a/tests/mmexternal-response-too-long.sh
+++ b/tests/mmexternal-response-too-long.sh
@@ -1,11 +1,37 @@
 #!/bin/bash
 # Regression coverage for bounded mmexternal replies: a helper that emits an
 # oversized unterminated JSON line must be restarted before responseTimeout is
-# reached, and the queue must continue to the following action. The oracle
-# waits a short time for both downstream messages and checks that a second
+# reached, and the queue must continue to the following action. The oracle uses
+# the helper side file to observe the oversized-reply and recovery milestones,
+# then checks that both messages reached the downstream action and that a second
 # helper instance handled the recovery message.
 
 . ${srcdir:=.}/diag.sh init
+
+wait_side_content() {
+	local needle="$1"
+	local timeout="${2:-${TB_TEST_TIMEOUT:-60}}"
+	local deadline=$(( $(date +%s) + timeout ))
+	local sidefile="$RSYSLOG_DYNNAME.side"
+
+	while true; do
+		if [ -f "$sidefile" ] && grep -Fq -- "$needle" "$sidefile"; then
+			printf 'side-file milestone reached: %s\n' "$needle"
+			return
+		fi
+		if [ "$(date +%s)" -ge "$deadline" ]; then
+			printf 'FAIL: side-file milestone not reached before timeout: %s\n' "$needle"
+			if [ -f "$sidefile" ]; then
+				printf 'side-file content:\n'
+				cat -n "$sidefile"
+			else
+				printf 'side-file %s does not exist\n' "$sidefile"
+			fi
+			error_exit 1
+		fi
+		$TESTTOOL_DIR/msleep 200
+	done
+}
 
 generate_conf
 add_conf '
@@ -25,8 +51,10 @@ if $msg contains "too-long-message" then {
 
 startup
 injectmsg literal "<13>Mar 10 01:00:00 host tag:too-long-message-1"
+wait_side_content "Oversized too-long-message-1"
 injectmsg literal "<13>Mar 10 01:00:00 host tag:too-long-message-2"
-wait_file_lines "$RSYSLOG_OUT_LOG" 2 5
+wait_side_content "Recovered too-long-message-2"
+wait_file_lines "$RSYSLOG_OUT_LOG" 2
 shutdown_when_empty
 wait_shutdown
 

